src/analysis/locally_convex/with_seminorms.lean

```diff
@@ -637,7 +639,78 @@ end
 
 end seminorm
 
-end continuous_bounded
+end continuous_of_bounded
+
+section bounded_of_countinuous
+
+namespace seminorm
+
+variables [nontrivially_normed_field 𝕜] [add_comm_group E] [module 𝕜 E]
+  [seminormed_add_comm_group F] [normed_space 𝕜 F]
+  {p : seminorm_family 𝕜 E ι}
+
+/-- In a semi-`normed_space`, a continuous seminorm is zero on elements of norm `0`. -/
+lemma map_eq_zero_of_norm_zero (q : seminorm 𝕜 F)
+  (hq : continuous q) {x : F} (hx : ‖x‖ = 0) : q x = 0 :=
+(map_zero q) ▸
+  ((specializes_iff_mem_closure.mpr $ mem_closure_zero_iff_norm.mpr hx).map hq).eq.symm
+
+/-- Let `F` be a semi-`normed_space` over a `nontrivially_normed_field`, and let `q` be a
+seminorm on `F`. If `q` is continuous, then it is uniformly controlled by the norm, that is there
+is some `C > 0` such that `∀ x, q x ≤ C * ‖x‖`.
+The continuity ensures boundedness on a ball of some radius `ε`. The nontriviality of the
+norm is then used to rescale any element into an element of norm in `[ε/C, ε[`, thus with a
+controlled image by `q`. The control of `q` at the original element follows by rescaling. -/
+lemma bound_of_continuous_normed_space (q : seminorm 𝕜 F)
+  (hq : continuous q) : ∃ C, 0 < C ∧ (∀ x : F, q x ≤ C * ‖x‖) :=
+begin
+  have hq' : tendsto q (𝓝 0) (𝓝 0) := map_zero q ▸ hq.tendsto 0,
+  rcases normed_add_comm_group.nhds_zero_basis_norm_lt.mem_iff.mp (hq' $ Iio_mem_nhds one_pos)
+    with ⟨ε, ε_pos, hε⟩,
+  rcases normed_field.exists_one_lt_norm 𝕜 with ⟨c, hc⟩,
+  have : 0 < ‖c‖ / ε := by positivity,
+  refine ⟨‖c‖ / ε, this, λ x, _⟩,
+  by_cases hx : ‖x‖ = 0,
+  { rw [hx, mul_zero],
+    exact le_of_eq (map_eq_zero_of_norm_zero q hq hx) },
+  refine (norm_seminorm 𝕜 F).bound_of_shell q ε_pos hc (λ x hle hlt, _) hx,
+  refine (le_of_lt $ hε hlt).trans _,
+  rwa [← div_le_iff' this, one_div_div]
+end
+
+/-- Let `E` be a topological vector `nontrivially_normed_field` whose topology is generated
```

Comment:
I don't understand this sentence. Did you mean "topological vector space over a nontrivially normed field"?

Reply:
Yes! :facepalm: